### ISSUE TITLE
Conflicts in dependency management

### ISSUE THREAD

**Issue by dsyer** [spring-cloud/spring-cloud-consul#282]
Build a simple cloud app with Maven 3.4, see the warnings, grep for "consul":

```
[WARNING] Dependency 'org.apache.httpcomponents:httpclient:jar' has conflicting dependency management in model 'org.springframework.cloud:spring-cloud-dependencies:pom:Dalston.BUILD-SNAPSHOT' ('org.springframework.cloud:spring-cloud-consul-dependencies:1.2.0.BUILD-SNAPSHOT', 'org.springframework.cloud:spring-cloud-vault-dependencies:1.0.0.BUILD-SNAPSHOT'). To resolve the conflicts, declare the dependency management for dependency 'org.apache.httpcomponents:httpclient:jar' directly in the dependency management of model 'org.springframework.cloud:spring-cloud-dependencies:pom:Dalston.BUILD-SNAPSHOT' to override what gets imported. If the Maven version in use supports it, add exclusions for the conflicting dependencies or use the include scope feature to rearrange the causing dependencies in the inheritance hierarchy applying standard override logic based on artifact coordinates. Without resolving the conflicts, your build relies on indeterministic behaviour. @ org.springframework.cloud:spring-cloud-dependencies:Dalston.BUILD-SNAPSHOT
[WARNING] Dependency 'org.apache.httpcomponents:httpclient:jar' has conflicting dependency management in model 'org.springframework.cloud.launcher:spring-cloud-launcher-deployer:jar:1.3.0.BUILD-SNAPSHOT' @ '/home/dsyer/dev/cloud/master/cli/spring-cloud-launcher/spring-cloud-launcher-deployer/pom.xml' ('org.springframework.boot:spring-boot-dependencies:1.5.1.RELEASE', 'org.springframework.cloud:spring-cloud-consul-dependencies:1.2.0.BUILD-SNAPSHOT'). To resolve the conflicts, declare the dependency management for dependency 'org.apache.httpcomponents:httpclient:jar' directly in the dependency management of model 'org.springframework.cloud.launcher:spring-cloud-launcher-deployer:jar:1.3.0.BUILD-SNAPSHOT' to override what gets imported. If the Maven version in use supports it, add exclusions for the conflicting dependencies or use the include scope feature to rearrange the causing dependencies in the inheritance hierarchy applying standard override logic based on artifact coordinates. Without resolving the conflicts, your build relies on indeterministic behaviour.
[WARNING] Dependency 'org.apache.httpcomponents:httpcore:jar' has conflicting dependency management in model 'org.springframework.cloud.launcher:spring-cloud-launcher-deployer:jar:1.3.0.BUILD-SNAPSHOT' @ '/home/dsyer/dev/cloud/master/cli/spring-cloud-launcher/spring-cloud-launcher-deployer/pom.xml' ('org.springframework.boot:spring-boot-dependencies:1.5.1.RELEASE', 'org.springframework.cloud:spring-cloud-consul-dependencies:1.2.0.BUILD-SNAPSHOT'). To resolve the conflicts, declare the dependency management for dependency 'org.apache.httpcomponents:httpcore:jar' directly in the dependency management of model 'org.springframework.cloud.launcher:spring-cloud-launcher-deployer:jar:1.3.0.BUILD-SNAPSHOT' to override what gets imported. If the Maven version in use supports it, add exclusions for the conflicting dependencies or use the include scope feature to rearrange the causing dependencies in the inheritance hierarchy applying standard override logic based on artifact coordinates. Without resolving the conflicts, your build relies on indeterministic behaviour.
[WARNING] Dependency 'com.google.code.gson:gson:jar' has conflicting dependency management in model 'org.springframework.cloud.launcher:spring-cloud-launcher-deployer:jar:1.3.0.BUILD-SNAPSHOT' @ '/home/dsyer/dev/cloud/master/cli/spring-cloud-launcher/spring-cloud-launcher-deployer/pom.xml' ('org.springframework.boot:spring-boot-dependencies:1.5.1.RELEASE', 'org.springframework.cloud:spring-cloud-consul-dependencies:1.2.0.BUILD-SNAPSHOT'). To resolve the conflicts, declare the dependency management for dependency 'com.google.code.gson:gson:jar' directly in the dependency management of model 'org.springframework.cloud.launcher:spring-cloud-launcher-deployer:jar:1.3.0.BUILD-SNAPSHOT' to override what gets imported. If the Maven version in use supports it, add exclusions for the conflicting dependencies or use the include scope feature to rearrange the causing dependencies in the inheritance hierarchy applying standard override logic based on artifact coordinates. Without resolving the conflicts, your build relies on indeterministic behaviour.
[WARNING] Dependency 'joda-time:joda-time:jar' has conflicting dependency management in model 'org.springframework.cloud.launcher:spring-cloud-launcher-deployer:jar:1.3.0.BUILD-SNAPSHOT' @ '/home/dsyer/dev/cloud/master/cli/spring-cloud-launcher/spring-cloud-launcher-deployer/pom.xml' ('org.springframework.boot:spring-boot-dependencies:1.5.1.RELEASE', 'org.springframework.cloud:spring-cloud-consul-dependencies:1.2.0.BUILD-SNAPSHOT'). To resolve the conflicts, declare the dependency management for dependency 'joda-time:joda-time:jar' directly in the dependency management of model 'org.springframework.cloud.launcher:spring-cloud-launcher-deployer:jar:1.3.0.BUILD-SNAPSHOT' to override what gets imported. If the Maven version in use supports it, add exclusions for the conflicting dependencies or use the include scope feature to rearrange the causing dependencies in the inheritance hierarchy applying standard override logic based on artifact coordinates. Without resolving the conflicts, your build relies on indeterministic behaviour.
```

**Comment by spencergibb** [comment]
@dsyer what's the fix for these?

**Comment by spencergibb** [comment]
from @dsyer 
> move it to the parent.
> [3:24]  
And then worry about why you need a different version than the one provided by Boot

**Comment by spencergibb** [comment]
Closed via 90eeebf3e29c9bb3bb3b72fa4d61339e9519fde2
